src/lib/context-menu.ts

```diff
@@ -255,6 +256,38 @@ export class ContextMenu {
         })
       )
     }
+    if (type === schema.nodes.table_element) {
+      let addGeneralNote = true
+      const tableElementFooter = findChildrenByType(
+        this.node,
+        schema.nodes.table_element_footer
+      )
+      if (tableElementFooter.length) {
```

Comment:
You can get rid of these nested if statements and directly assign the value to a const.
i.e
```
const hasGeneralNote = tableElementFooter.length && tableElementFooter[0].node.firstChild?.type === schema.nodes.paragraph

if (!hasGeneralNote) {
menu.appendChild() ..
}
```

Reply:
done